core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -420,21 +430,54 @@ protected static void assertValidate(QueryBuilder queryBuilder, int totalExpecte
 
     /**
      * create a random value for either {@link BaseQueryTestCase#BOOLEAN_FIELD_NAME}, {@link BaseQueryTestCase#INT_FIELD_NAME},
-     * {@link BaseQueryTestCase#DOUBLE_FIELD_NAME} or {@link BaseQueryTestCase#STRING_FIELD_NAME}, or a String value by default
+     * {@link BaseQueryTestCase#DOUBLE_FIELD_NAME}, {@link BaseQueryTestCase#STRING_FIELD_NAME} or
+     * {@link BaseQueryTestCase#DATE_FIELD_NAME}, or a String value by default
      */
-    protected static Object randomValueForField(String fieldName) {
+    protected static Object getRandomValueForFieldName(String fieldName) {
         Object value;
         switch (fieldName) {
-            case BOOLEAN_FIELD_NAME: value = randomBoolean(); break;
-            case INT_FIELD_NAME: value = randomInt(); break;
-            case DOUBLE_FIELD_NAME: value = randomDouble(); break;
-            case STRING_FIELD_NAME: value = randomAsciiOfLengthBetween(1, 10); break;
-            default : value = randomAsciiOfLengthBetween(1, 10);
+            case STRING_FIELD_NAME:
+                value = rarely() ? randomUnicodeOfLength(10) : randomAsciiOfLengthBetween(1, 10); // unicode in 10% cases
+                break;
+            case INT_FIELD_NAME:
+                value = randomIntBetween(0, 10);
+                break;
+            case DOUBLE_FIELD_NAME:
+                value = randomDouble() * 10;
+                break;
+            case BOOLEAN_FIELD_NAME:
+                value = randomBoolean();
+                break;
+            case DATE_FIELD_NAME:
+                value = new DateTime(System.currentTimeMillis(), DateTimeZone.UTC).toString();
+                break;
+            default:
+                value = randomAsciiOfLengthBetween(1, 10);
         }
         return value;
     }
 
     /**
+     * Helper method to return a mapped or a random field
+     */
+    protected String getRandomFieldName() {
+        // if no type is set then return a random field name
+        if (currentTypes == null || currentTypes.length == 0 || randomBoolean()) {
+            return randomAsciiOfLengthBetween(1, 10);
+        }
+        // handle exclude fields
+        return randomFrom(STRING_FIELD_NAME, INT_FIELD_NAME, BOOLEAN_FIELD_NAME, BOOLEAN_FIELD_NAME, DATE_FIELD_NAME);
```

Comment:
The problem is that some queries do not support all field types. Since we removed the exclude parameter, I think it's better to keep this to be an expected set of types, instead of returning all types, even those which we may add in the future.